tracing-subscriber/src/layer/mod.rs

```diff
@@ -393,6 +393,38 @@
 //! # Ok(()) }
 //! ```
 //!
+//! ## Runtime-configurable Layers
+//!
+//! Conditionally enabling/disabling layers at runtime can be challenging due to the generics
+//! used Layers implementations. However, A Layer wrapped in an `Option` is also Layers, which
+//! allows some runtime configuration:
```

Comment:
some copyedits --- how about something like
```suggestion
//! In some cases, a particular [`Layer`] may be enabled or disabled based on
//! runtime configuration. This can introduce challenges, because the type of a
//! layered [`Subscriber`] depends on which layers are added to it: if an `if`
//! or `match` expression adds some [`Layer`]s in one branch and other layers
//! in another, the [`Subscriber`] values returned by those branches will have
//! different types.
//!
//! However, a [`Layer`] wrapped in an [`Option`] also implements the [`Layer`]
//! trait. This allows individual layers to be enabled or disabled at runtime
//! while always producing a [`Subscriber`] of the same type. For example:
```

Reply:
also, i think it _might_ be nice to introduce these problems with a `compile_fail` example that shows code that _doesn't_ work because of creating subscribers of two different types...but it's up to you if you want to spend time on that...

Reply:
yeah, those copy edits are way better. sorry, when i wrote this, i felt myself fading terribly, and this much better.

I'll also add a `compile_fail` example.